### DISCUSSION TITLE
Идея по устранению полной ретрансмиссии

### DISCUSSION THREAD

**Discussion by EvgenKo423** [hufrea/byedpi#122] [Ideas]
Есть идея как побороть полную ретрансмиссию на Windows и, возможно, на BSD тоже:
Она происходит из-за того, что ОС объединяет все пакеты с одинаковым ACK'ом, а в случае с `--fake` она каждый раз перечитывает файл заново, что позволяет заменить пакет, верно?
Можно для отправки 2-го пакета использовать 2-й файл, а после отправки обнулять его размер. Тогда ОС не сможет его перечитать и отправит только 1-ю часть! (по идее)

Желательно от неё избавиться, т.к. DPI может её обработать и это всё-таки лишний трафик.

**Comment by hufrea** [comment]
> Можно для отправки 2-го пакета использовать 2-й файл, а после отправки обнулять его размер.

Это не сработает, ОС все равно отправит данные. В Windows, видимо, усечение файла не затрагивает файловый кеш, Linux же просто отправляет нули в таком случае. Да и если бы ядро могло бы позволить уменьшить TCP SEQ, это было бы очень странно, ведь это протокол для потоковой передачи.
